src/strip.rs

```diff
@@ -118,24 +121,95 @@ fn get_mod<F: Write>(current: &Option<TypeStruct>, out_file: &mut F) -> bool {
     }
 }
 
+enum BlockKind<'a> {
+    Comment((String, String, &'a str)),
+    DocComment((String, String, &'a str)),
+    Other(&'a str),
+}
+
+fn get_three_parts<'a>(before: &'a str, after: &'a str, stop: &str) -> (String, String, &'a str) {
+    if let Some(pos) = after.find(stop) {
+        (before.to_owned(), after[0..pos].to_owned(), &after[pos..])
+    } else {
+        (before.to_owned(), after.to_owned(), &after[after.len() - 1..])
+    }
+}
+
+fn find_one_of<'a>(comments: &[&str], doc_comments: &[&str], text: &'a str) -> BlockKind<'a> {
+    let mut last_pos = 0;
+
+    loop {
+        let tmp_text = &text[last_pos..];
+        if let Some(pos) = tmp_text.find('/') {
+            last_pos = pos + last_pos;
+            for com in doc_comments {
+                if tmp_text.starts_with(com) {
+                    if &com[1..2] == "*" {
+                        return BlockKind::DocComment(get_three_parts(&text[0..last_pos], &text[last_pos..], "*/"))
+                    } else {
+                        return BlockKind::DocComment(get_three_parts(&text[0..last_pos], &text[last_pos..], "\n"))
+                    }
+                }
+            }
+            for com in comments {
+                if tmp_text.starts_with(com) {
+                    if &com[1..2] == "*" {
+                        return BlockKind::Comment(get_three_parts(&text[0..last_pos], &text[last_pos..], "*/"))
+                    } else {
+                        return BlockKind::Comment(get_three_parts(&text[0..last_pos], &text[last_pos..], "\n"))
+                    }
+                }
+            }
+        } else {
+            return BlockKind::Other(text)
+        }
+    }
+}
+
+fn transform_code(code: &str) -> String {
+    code.replace("{", " { ")
+        .replace("}", " } ")
+        .replace("///", "/// ")
+        .replace("//!", "//! ")
+        .replace("/*!", "/*! ")
+        .replace(":", " : ")
+        .replace(" :  : ", "::")
+        .replace("*/", " */")
+        .replace("\n", " \n ")
+        .replace("!(", " !! (")
+        .replace("!  {", " !? {")
+        .replace(",", ", ")
+        .replace("(", " (")
+}
+
+fn clean_input(mut s: &str) -> String {
+    let mut ret = String::new();
+    loop {
+        s = match find_one_of(&vec!("//", "/*"), &vec!("///", "/*!", "//!"), s) {
+            BlockKind::Comment((s, comment, after)) => {
+                ret.push_str(&transform_code(&s));
+                ret.push_str(&iter::repeat(" \n ").take(comment.split("\n").collect::<Vec<&str>>().len()).collect::<Vec<&str>>().join(""));
```

Comment:
Every `collect` is allocating a vector. If you want to preallocate the correct length you can use `reserve` and `with_capacity` but that makes sense only after you stop creating transient vectors left and right.